File: Create/Factory_Method/Factory_Method.java
// Product
public interface Document {
    void open();
}

// ConcreteProduct
public class DrawingDocument implements Document {
    @Override
    public void open() {
        System.out.println("Drawing document opened");
    }
}

// Creator
public abstract class Application {
    public void openDocument() {
        Document doc = createDocument();
        doc.open();
    }
    protected abstract Document createDocument();
}

// ConcreteCreator
public class DrawingApplication extends Application {
    @Override
    protected Document createDocument() {
        return new DrawingDocument();
    }

    public static void main(String[] args) {
        new DrawingApplication().openDocument();
    }
}

File: Create/Factory_Method/Factory_Method.py
from abc import ABC, abstractmethod

# Product
class Document(ABC):
    @abstractmethod
    def open(self) -> None: ...

# ConcreteProduct
class DrawingDocument(Document):
    def open(self) -> None:
        print("Drawing document opened")

# Creator
class Application(ABC):
    def open_document(self) -> None:
        doc = self.create_document()
        doc.open()

    @abstractmethod
    def create_document(self) -> Document: ...

# ConcreteCreator
class DrawingApplication(Application):
    def create_document(self) -> Document:
        return DrawingDocument()

if __name__ == "__main__":
    DrawingApplication().open_document()